client/src/main/java/io/pravega/client/connection/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -27,19 +27,26 @@
  * synchronous writers. Otherwise the batch size is set to the amount of data that will be written in the next
  * {@link #MAX_BATCH_TIME_MILLIS} or half the server round trip time (whichever is less)
  */
-class AppendBatchSizeTrackerImpl implements AppendBatchSizeTracker {
-    private static final int MAX_BATCH_TIME_MILLIS = 100;
-
+public class AppendBatchSizeTrackerImpl implements AppendBatchSizeTracker {
+    private static final double NANOS_PER_MILLI = 1000000;
+    
+    // This must be less than WireCommands.MAX_WIRECOMMAND_SIZE / 2;
+    private static final int MAX_BATCH_SIZE = initializeConstant("PRAVEGA_MAX_BATCH_SIZE",
+                                                                 2 * TcpClientConnection.TCP_BUFFER_SIZE - 1024);
+    private static final int BASE_TIME_NANOS = initializeConstant("PRAVEGA_BATCH_BASE_TIME_NANOS", 500000);
+    private static final int BASE_SIZE = initializeConstant("PRAVEGA_BATCH_BASE_SIZE", 0);
+    private static final double OUTSTANDING_FRACTION = 1.0 / initializeConstant("PRAVEGA_BATCH_OUTSTANDING_DENOMINATOR", 2);
+    
     private final Supplier<Long> clock;
     private final AtomicLong lastAppendNumber;
     private final AtomicLong lastAppendTime;
     private final AtomicLong lastAckNumber;
-    private final ExponentialMovingAverage eventSize = new ExponentialMovingAverage(1024, 0.1, true);
-    private final ExponentialMovingAverage millisBetweenAppends = new ExponentialMovingAverage(10, 0.1, false);
-    private final ExponentialMovingAverage appendsOutstanding = new ExponentialMovingAverage(2, 0.05, false);
+    private final ExponentialMovingAverage eventSize = new ExponentialMovingAverage(1024, 0.01, true);
+    private final ExponentialMovingAverage nanosBetweenAppends = new ExponentialMovingAverage(10 * NANOS_PER_MILLI, 0.001, true);
```

Comment:
This switch from _millis_ to _nanos_ seems unrelated to the removal of Netty.
It is required for this patch to work ?

Reply:
Nanos vs millis is not requred, but it did give bad results which interfered with testing. (It turns out it's fairly common for the difference in time in millis to be "0".)